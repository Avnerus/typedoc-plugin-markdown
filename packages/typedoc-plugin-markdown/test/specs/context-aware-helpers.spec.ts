import * as Handlebars from 'handlebars';

import { TestApp } from '../test-app';

describe(`ContextAwareHelpers:`, () => {
  let testApp: TestApp;

  beforeAll(() => {
    testApp = new TestApp(['breadcrumbs.ts']);
  });

  describe(`(defaults)`, () => {
    beforeAll(() => {
      testApp.bootstrap({});
    });

    test(`should set 'ifShowNamedAnchors' to 'false' by default`, () => {
      expect(
        Handlebars.helpers.ifShowNamedAnchors(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should set 'ifShowBreadcrumbs' to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowBreadcrumbs(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set 'ifShowProjectName' to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowProjectName(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set 'ifShowReflectionTitle' to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowReflectionTitle(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set 'ifShowReflectionPath' to 'false' by default`, () => {
      expect(
        Handlebars.helpers.ifShowReflectionPath(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should set 'ifShowIndex' to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowIndex(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should compile relativeURL helper`, () => {
      const url = testApp.findReflection('Breadcrumbs').url;
      expect(Handlebars.helpers.relativeURL(url)).toMatchSnapshot();
    });
  });

  describe(`(with plugin options)`, () => {
    beforeAll(() => {
      testApp.bootstrap({
        publicPath: 'test-public-path',
        namedAnchors: true,
        hideBreadcrumbs: true,
        hideProjectName: true,
      });
    });

    test(`should set named anchors to 'true'`, () => {
      expect(
        Handlebars.helpers.ifShowNamedAnchors(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set breadcrumbs to 'false'`, () => {
      expect(
        Handlebars.helpers.ifShowBreadcrumbs(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should set project name to 'false'`, () => {
      expect(
        Handlebars.helpers.ifShowProjectName(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should compile relativeURL helper with public path`, () => {
      const url = testApp.findReflection('Breadcrumbs').url;
      expect(Handlebars.helpers.relativeURL(url)).toMatchSnapshot();
    });
  });
  /*
  describe(`(ifShowIndex helper default)`, () => {
    beforeAll(() => {
      testApp = new TestApp(['breadcrumbs.ts']);
      testApp.bootstrap();
    });
    test(`should always be true`, () => {
      const reflection = testApp.findReflection('Breadcrumbs');
      expect(
        Handlebars.helpers.ifShowIndex.call(
          reflection,
          TestApp.handlebarsOptionsStub,
        ),
      ).toBeTruthy();
    });
  });

  describe(`(ifShowIndex helper with navigation)`, () => {
    beforeAll(() => {
      testApp = new TestApp(['breadcrumbs.ts']);
      const customThemePath = path.resolve(
        __dirname,
        '..',
        'stubs',
        'custom-theme',
      );
      testApp.bootstrap({
        theme: customThemePath,
      });
    });
    test(`should be false on classes if navigation enabled`, () => {
      const reflection = testApp.findReflection('Breadcrumbs');
      expect(
        Handlebars.helpers.ifShowIndex.call(
          reflection,
          TestApp.handlebarsOptionsStub,
        ),
      ).toBeFalsy();
    });
  });*/
});
