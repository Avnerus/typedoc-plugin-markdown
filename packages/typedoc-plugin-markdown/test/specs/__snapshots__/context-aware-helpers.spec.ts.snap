// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextAwareHelpers: (defaults) should compile relativeURL helper 1`] = `"breadcrumbs.md"`;

exports[`ContextAwareHelpers: (with plugin options) should compile relativeURL helper with public path 1`] = `"test-public-pathclasses/breadcrumbs.md"`;
